src/runtime/methodbinder.cs

```diff
@@ -415,11 +416,42 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                 }
                 else
                 {
-                    op = (arrayStart == paramIndex)
-                        // map remaining Python arguments to a tuple since
-                        // the managed function accepts it - hopefully :]
-                        ? Runtime.PyTuple_GetSlice(args, arrayStart, pyArgCount)
-                        : Runtime.PyTuple_GetItem(args, paramIndex);
+                    if(arrayStart == paramIndex)
+                    {
+                        // for a params method, we may have a sequence or single/multiple items
+                        // here we look to see if the item at the paramIndex is there or not
+                        // and then if it is a sequence itself.
+                        if((pyArgCount - paramIndex) == 1)
+                        {
+                            // we only have one argument left, so we need to check it
+                            // to see if it is a sequence or a single item
+                            IntPtr item = Runtime.PyTuple_GetItem(args, paramIndex);
+                            if(!Runtime.PyString_Check(item) && Runtime.PySequence_Check(item))
+                            {
+                                // it's a sequence (and not a string), so we use it as the op
+                                op = item;
+                            }
+                            else
+                            {
+                                doDecref = true;
+                                op = Runtime.PyTuple_GetSlice(args, arrayStart, pyArgCount);
+                                if (item != IntPtr.Zero)
+                                {
+                                    Runtime.XDecref(item);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            doDecref = true;
+                            op = Runtime.PyTuple_GetSlice(args, arrayStart, pyArgCount);
+                        }                                         
```

Comment:
Just a suggestion: maybe move params argument handling to a separate function for clarity

Reply:
That's a good idea. Would you like it done as part of this PR, or a separate PR?